### PR TITLE
JAMES-3087 Sequentially process authentication strategies

### PR DESCRIPTION
**Pull request by chibenwa**
Solutions relying on JWT were also execution standard JMAP authentication

This is because flatMap is eager.

Sequentially processing avoids verbose logs of invalid access token format.

(behaviouyr pre-reactor-netty migration)

### CODE REVIEW

**Comment by mbaechler** [comment]
could you add a test?

**Comment by chibenwa** [comment]
> could you add a test?

Which kind of tests?

A test to ensure a log is not fired? We never write tests with logs.

A test with a mock? -> I'm struggling so much recently with our limited mock test suite recently that I must admit being really reluctant to continue extending it. Especially for a simple flatMap -> concatMap change.

Maybe a comment in the code could explain the intent and avoid future non intentional modification?

**Comment by mbaechler** [comment]
you can provide a throwing implementation and check that it's never called by verifying we don't generate an error

File: server/protocols/jmap-draft/src/test/java/org/apache/james/jmap/http/AuthenticationReactiveFilterTest.java [hunk 1]
```diff
@@ -81,7 +82,21 @@ public void filterShouldReturnUnauthorizedOnNullAuthorizationHeader() {
         assertThatThrownBy(() -> testee.authenticate(mockedRequest).block())
             .isInstanceOf(UnauthorizedException.class);
     }
-    
+
+    @Test
+    public void authenticationStrategiesShouldNotBeEagerlySubScribed() {
```
Comment (remk) [inline server/protocols/jmap-draft/src/test/java/org/apache/james/jmap/http/AuthenticationReactiveFilterTest.java]: s/SubScribed/Subscribed

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**